src/vcpkg/dependencies.cpp

```diff
@@ -1551,79 +1551,79 @@ namespace vcpkg::Dependencies
                                                          const Version& version,
                                                          const std::string& origin)
         {
-            ExpectedS<const vcpkg::SourceControlFileAndLocation&> maybe_scfl;
-
             // if this port is an overlay port, ignore the given version and use the version from the overlay
             auto maybe_overlay = m_o_provider.get_control_file(graph_entry.first.name());
-            if (auto p_overlay = maybe_overlay.get())
+            const vcpkg::SourceControlFileAndLocation* p_scfl = maybe_overlay.get();
+            if (p_scfl)
             {
-                const auto overlay_version = p_overlay->source_control_file->to_version();
+                const auto overlay_version = p_scfl->source_control_file->to_version();
                 // If the original request did not match the overlay version, restart this function to operate on the
                 // overlay version
                 if (version != overlay_version)
                 {
-                    return require_port_version(graph_entry, overlay_version, origin);
+                    require_port_version(graph_entry, overlay_version, origin);
+                    return;
                 }
-                maybe_scfl = *p_overlay;
             }
             else
             {
                 // if there is a override, ignore the given version and use the version from the override
                 auto over_it = m_overrides.find(graph_entry.first.name());
                 if (over_it != m_overrides.end() && over_it->second != version)
                 {
-                    return require_port_version(graph_entry, over_it->second, origin);
+                    require_port_version(graph_entry, over_it->second, origin);
+                    return;
                 }
-                maybe_scfl = m_ver_provider.get_control_file({graph_entry.first.name(), version});
-            }
 
-            if (auto p_scfl = maybe_scfl.get())
-            {
-                auto& versioned_graph_entry = graph_entry.second.emplace_node(
-                    p_scfl->source_control_file->core_paragraph->version_scheme, version);
-                versioned_graph_entry.origins.push_back(origin);
-                // Use the new source control file if we currently don't have one or the new one is newer
-                bool replace;
-                if (versioned_graph_entry.scfl == nullptr)
-                {
-                    replace = true;
-                }
-                else if (versioned_graph_entry.scfl == p_scfl)
+                const auto maybe_scfl = m_ver_provider.get_control_file({graph_entry.first.name(), version});
+                p_scfl = maybe_scfl.get();
+                if (!p_scfl)
                 {
-                    replace = false;
+                    m_errors.push_back(maybe_scfl.error());
```

Comment:
This could be a move if `maybe_scfl` wasn't const.

Reply:
Fixed!

Reply:
But (now) it still pushes the error, not the "moved" object?

Reply:
@dg0yt: It's still pushing a moved object; that calls


https://github.com/microsoft/vcpkg-tool/blob/01294638b93e0d472007491d449db1ab7cdf3aa4/include/vcpkg/base/expected.h#L182